seccomp/pidTracker.go

```diff
@@ -0,0 +1,94 @@
+//
+// Copyright 2019-2022 Nestybox, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package seccomp
+
+import (
+	"sync"
+)
+
+type pidData struct {
+	refcnt int
+	mu     sync.Mutex
+}
+
+type seccompNotifPidTracker struct {
+	mu       sync.Mutex
```

Comment:
Based on Lock() and Unlock() implementations, this mutex should be a RW one to reduce contention to the minimum.

Reply:
Good idea, fixed.